python/pyarrow/array.pxi

```diff
@@ -2747,69 +2824,8 @@ cdef class ListViewArray(Array):
         """
         return pyarrow_wrap_array((<CListViewArray*> self.ap).sizes())
 
-    def flatten(self, memory_pool=None):
```

Comment:
Pyarrow's mainly changes should be here. `[Large]ListViewArray` does not seem to need to maintain an independent flatten interface separately, and can directly use the one provided by `BaseListArray`.